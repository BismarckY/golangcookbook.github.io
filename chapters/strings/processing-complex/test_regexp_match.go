package main

import (
	"fmt"
	"regexp"
)

func main() {
	s := "[one][two][three]"
	matches := regexp.MustCompile(`\[(.*?)\]`).FindAllStringSubmatch(s, -1)
	if matches == nil {
		fmt.Println("No matches found.")
		return
	}

	for i, match := range matches {
		full := match[0]
		submatches := match[1:len(match)]
		fmt.Printf("%v => \"%v\" from \"%v\"\n", i, submatches[0], full)
	}
}
