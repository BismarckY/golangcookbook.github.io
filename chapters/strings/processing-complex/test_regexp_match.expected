0 => "one" from "[one]"
1 => "two" from "[two]"
2 => "three" from "[three]"
